### ISSUE TITLE
Low training accuracy and Errors when test videos 

### ISSUE THREAD

**Issue by Xiao-G1023** [yjxiong/temporal-segment-networks#38]
Hi, Mr. Xiong!
1. Now I can train the TSN on the dataset hmdb51_split_1_rgb with the default parameters using `bash scripts/train_tsn.sh ucf101 rgb` . After 2500 iteration, the train net loss = 0.17426, the test accuracy = 0.530679 and test loss = 2.30292. 
### Which parameters may affect this test accuracy, i.e., how could I get a similar accuracy as your work?
2. I met some problems when I test the TSN Network. As the file Readme, I input the code
`python tools/eval_net.py hmdb51 1 rgb /raid/Public/zhoup_DS/HMDB51/TSN_hmdb51_frames/  models/hmdb51/tsn_bn_inception_rgb_deploy.prototxt models/hmdb51_split_1_tsn_rgb_reference_bn_inception.caffemodel --num_worker 3 --save_scores ZhouExample/hmdb51_split1_test.txt`
#### However, it goes wrong with the following output:
....
6400 videos parsed
6600 videos parsed
frame folder analysis done
[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 19:12: Message type "caffe.LayerParameter" has no field named "bn_param".
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1216 09:33:18.682155 27370 upgrade_proto.cpp:79] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: models/hmdb51/tsn_bn_inception_rgb_deploy.prototxt
*** Check failure stack trace: ***
[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 19:12: Message type "caffe.LayerParameter" has no field named "bn_param".
......
*** Check failure stack trace: ***
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1216 09:35:24.345011 27464 common.cpp:169] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
....

##### When I check the cuda using `TSN_master/lib/caffe-action/build/install/bin$ ldd caffe | grep cuda` , it shows like 
libcudart.so.7.5 => /usr/local/cuda/lib64/libcudart.so.7.5 (0x00007f387598e000)
libcurand.so.7.5 => /usr/local/cuda/lib64/libcurand.so.7.5 (0x00007f3872125000)
libcublas.so.7.5 => /usr/local/cuda/lib64/libcublas.so.7.5 (0x00007f3870846000)
libcudnn.so.5 => /usr/local/cuda/lib64/libcudnn.so.5 (0x00007f386ccfb000)

### Have you met the errors? Could you give me some advice about that? 
Thank you and looking for your reply. Best wishes!

**Comment by yjxiong** [comment]
1. See this 

https://github.com/yjxiong/temporal-segment-networks/wiki/Frequently-Asked-Questions#testing-tsn

2. You are using another version of Caffe in your testing scripts. Please use the bundled Caffe for the Python scripts.

**Comment by Xiao-G1023** [comment]
Thanks for reply! The Caffe version is downloaded in your github https://github.com/yjxiong/caffe and carried out according to the build_all.sh. I don't know whether the "bundled Caffe" is this one. 
Thanks~

**Comment by yjxiong** [comment]
Look at this sentence in you log
```
19:12: Message type "caffe.LayerParameter" has no field named "bn_param".
```

This means the underlying Caffe libraries you are using in your Python scripts do not have the `BNLayer` in the bundled Caffe. This means you are linking to another version of Caffe installed on your machine.

**Comment by Xiao-G1023** [comment]
According to your advice, in the file ~/.bashrc , I add
 `PYTHONPATH= /home/zhoupeipei/action-code/2016ECCV_TSN_master/lib/caffe-action/python:$PYTHONPATH` and
 `LD_LIBRARY_PATH=/home/zhoupeipei/action-code/2016ECCV_TSN_master/lib/caffe-action/build/lib:$LD_LIBRARY_PATH`.
#### After I rebuild the `caffe-action` based on `build_all.sh`, the errors disappear when I test videos again! 
Thanks for your patience! It is very kind of you!

**Comment by yjxiong** [comment]
Looks good, normally the test scripts will then report the results shown in the paper.

**Comment by yjxiong** [comment]
@Xiao-G1023 
have you got the correct results?

**Comment by Xiao-G1023** [comment]
@yjxiong Dear Mr. Xiong, now I could test and train RGB_split_1 of hmdb51.  I got the rgb_score_file and the flow_score_file as ReadMe file. Here, I found that the `SCORE_FILE` is a file_name string instead of a directory path and the test result is a `.zip` files.
At last, using `python tools/eval_scores.py RGB_SCORE_FILE.zip FLOW_SCORE_FILE.zip --score_weights 1 1.5`, I obtain the accuracy of  hmdb51_split1 is 69.477124%.
Thanks for your attention~  Best wishes to you and to my experiments.

**Comment by yjxiong** [comment]
The score file is actually storing two numpy arrays in a compressed way. 

See here for how they are stored
https://github.com/yjxiong/temporal-segment-networks/blob/master/tools/eval_net.py#L136

Currently, you can look at the `tools/eval_net.py` for ideas on how to analyze single videos. The script for this purpose is still on its way.

**Comment by Xiao-G1023** [comment]
OK! Thanks a lot! I am very grateful for your kindly help! 
祝熊老师及您的团队一切顺利，更上一层楼！

**Comment by yjxiong** [comment]
Good luck with your experiments!
Closing this. Please feel free to reopen it if you meet any further problem.

**Comment by actionderkokon** [comment]
hello，i met with the same problem，and set the LD_LIBRARY_PATH and PYTHONPATH environmental variables. However,when i rebuild caffe-action, a new problem:'no module named caffe', do u met the problem?
